database/gdb/gdb_model_cache.go

```diff
@@ -140,6 +172,16 @@ func (m *Model) saveSelectResultToCache(
 	if errCache := cacheObj.Set(ctx, cacheKey, cacheItem, m.cacheOption.Duration); errCache != nil {
 		intlog.Errorf(ctx, `%+v`, errCache)
 	}
+
+	if m.cacheEnabled && len(m.cacheOption.Group) > 0 {
+		keyGroup, errCache := cacheObj.Get(ctx, m.cacheOption.Group)
+		if errCache == nil {
+			keys := append(keyGroup.Strings(), cacheKey)
+			if errCache = cacheObj.Set(ctx, m.cacheOption.Group, keys, m.cacheOption.Duration); errCache != nil {
```

Comment:
The `Get` and `Set` function is not atomic, which will cause concurrent safety issue.

Reply:
You don't have to worry about concurrency security, and there's no problem with cache updates overwriting the original value

Reply:
@z5830865 Would it loose keys when there were multiple goroutines doing `Get` and `Set`?

Reply:
It is not good design using `Get/Set` to maintain the keys, as there's concurrent issue here.